### PR TITLE
✨ Expose Cache/Client options on Cluster

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@redhat.com>

### CODE REVIEW

**Comment by vincepri** [comment]
/hold

for tests

**Comment by vincepri** [comment]
@alvaroaleman @sbueringer thoughts on the approach?

File: pkg/cluster/cluster.go [hunk 1]
```diff
@@ -171,36 +196,63 @@ func New(config *rest.Config, opts ...Option) (Cluster, error) {
...
+		}
+		if clientOpts.HTTPClient == nil {
+			clientOpts.HTTPClient = options.HTTPClient
+		}
+		if clientOpts.Cache == nil {
+			clientOpts.Cache.Reader = cache
+			clientOpts.Cache.DisableFor = options.ClientDisableCacheFor
+			clientOpts.Cache.Unstructured = false
+		}
+		if !clientOpts.DryRun && options.DryRunClient {
```
Comment (alvaroaleman) [inline pkg/cluster/cluster.go:239]: maybe make `clientOpts.DryRun` a pointer to be able to differentiate between unset and set here?

File: pkg/cluster/cluster.go [hunk 2]
```diff
@@ -171,36 +196,63 @@ func New(config *rest.Config, opts ...Option) (Cluster, error) {
...
-		},
-	})
+	// Create the client, and default its options.
+	clientOpts := options.Client
+	{
+		if clientOpts.Scheme == nil {
+			clientOpts.Scheme = options.Scheme
+		}
+		if clientOpts.Mapper == nil {
+			clientOpts.Mapper = mapper
```
Comment (alvaroaleman) [inline pkg/cluster/cluster.go:229]: Nit: Maybe also deprecate the `mapperProvider` and let the client itself default the `Mapper` if not passed?

**Review by alvaroaleman — COMMENTED** [review]
I like it a lot, much cleaner :)

File: pkg/cluster/cluster.go (hunk 1 above)
Comment (vincepri) [reply pkg/cluster/cluster.go]: Good idea, done!

File: pkg/cluster/cluster.go (hunk 2 above)
Comment (vincepri) [reply pkg/cluster/cluster.go:232]: Done!

Comment (vincepri) [reply pkg/cluster/cluster.go:232]: Both the cache and the client already default it, so we should be good there once we remove the mapperProvider

**Comment by sbueringer** [comment]
Looks good to me

/assign @alvaroaleman

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**